### ISSUE TITLE
Can not run the watchOS app on the emulator

### ISSUE THREAD

**Issue by StefanLobbenmeier** [joreilly/PeopleInSpace#12]
Reproduction steps:
1. clone the repository
2. open .xcworkspace in Xcode
3. change signing to local team
4. attempt to run on the local emulator

This is the output I get:

```
ld: warning: ignoring file /Volumes/macApps/fhswf/S5/Projekt/PeopleInSpace/common/build/cocoapods/framework/common.framework/common, building for watchOS Simulator-x86_64 but attempting to link with file built for watchOS Simulator-i386
Undefined symbols for architecture x86_64:
  "_OBJC_CLASS_$_CommonKoinKt", referenced from:
      objc-class-ref in ExtensionDelegate.o
  "_OBJC_CLASS_$_CommonAssignment", referenced from:
      objc-class-ref in ViewModel.o
  "_OBJC_CLASS_$_CommonPeopleInSpaceRepository", referenced from:
      objc-class-ref in ContentView.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

```

I guess the warning 
> watchOS Simulator-x86_64 but attempting to link with file built for watchOS Simulator-i386
already tells a lot but I am not sure how to change that

**Comment by StefanLobbenmeier** [comment]
My environment: 
macOS Catalina 10.15.4
Xcode 12.0.1

**Comment by joreilly** [comment]
@nealsanche just in case you've come across this?

**Comment by nealsanche** [comment]
Yeah, I have come across this, and it usually involves editing the target architectures until it works. I really don't know why it does this. I'll spend a bit of time and write down how I get it working, right now.

**Comment by nealsanche** [comment]
Okay, that took a few hours, messing around to try and get a minimal change that will allow compiling for the watchos simulator. It seems to be compiling for ARM64 simulators by default, and that really doesn't work at all, since the KMP module (common) gets compiled for i386 and then it'll skip linking it.

What did I do to get it working?

1) Click on PeopleInSpaceWatch in the left nav in xcode.
2) Tap build settings
3) Tap the PeopleInSpace Watchkit Extension target
4) Find the 'Architectures' and notice that it is defaulting to 'Standard Architectures'.
5) Click and edit that row, and choose Other...
6) Click $(ARCHS_STANDARD) and tap the - button on the bottom to remove it.
7) Hit the + button, and add 'i386'. I've tried 'x86_64' but that doesn't work for me. i386 will produce a warning about that architecture being deprecated, but it will run on the simulator.

Now a build should finish and install.

Really not sure how to make this work for both simulator and real hardware devices without having to edit these. Absolutely no idea. Maybe someone more familiar can fill me in on how things should be done.

Hope this helps.

**Comment by StefanLobbenmeier** [comment]
Thanks,  this did fix it! Just wanted to also leave a screenshot here for anyone facing the same issue with the steps you described.

![Screenshot 2020-10-07 at 10 30 28](https://user-images.githubusercontent.com/36509607/95307099-84b59800-0888-11eb-94cf-7d5a79af1968.png)

> Really not sure how to make this work for both simulator and real hardware devices without having to edit these. 

That means this can not be fixed for everyone by changing the code in the repository?

**Comment by StefanLobbenmeier** [comment]
Not sure if I should open another ticket or if its the same cause, but neither Standard architectures or i386 allow me to see the live preview in Xcode:
![image](https://user-images.githubusercontent.com/36509607/95309149-0c9ca180-088b-11eb-95de-eba41cd857f5.png)

**Comment by RdeWilde** [comment]
I'm still fighting this error. I changed the architecture to i386 as suggested, but now I get this error:

![image](https://user-images.githubusercontent.com/184016/115126749-d6767200-9fd1-11eb-95c8-f13c240952d8.png)

**Comment by RdeWilde** [comment]
@nealsanche did you happen to solve the above error? I've been hurting my head on this.

**Comment by StefanLobbenmeier** [comment]
Sorry still not much of an iOS / watchOS Developer, so not sure how to fix the above. One thing you could try is to checkout the commit that was current at that time and try again.

> I'm on an older iMac.

that really does not help much - what is your macOS version?

**Comment by RdeWilde** [comment]
Catalina and the newest xcode.

I think it is because WatchOS recently switched to a different architecture for the simulator, and it seems this is added to Kotlin 1.5.0. So updating might help, but I'm not 100% sure yet, as I'm still in the progress of fixing the rest of the bugs I created :)

**Comment by StefanOltmann** [comment]
On my Apple M1 MacBook I have a very similar problem and struggle to get it working.

The architecture for the simulator is here `arm64` opposed to Intel for older MacBooks.

I changed it in the `build.gradle.kts` by removing the now unnecessary if-else and adjusted some settings in the XCode project, but that didn't work.

I still get
`ld: in /Users/stefan/IdeaProjects/PeopleInSpace/common/build/cocoapods/framework/common.framework/common(result.o), building for iOS Simulator, but linking in object file built for iOS, file '/Users/stefan/IdeaProjects/PeopleInSpace/common/build/cocoapods/framework/common.framework/common' for architecture arm64`

I've limited experience in editing XCode settings, so I'm not sure how to change the project to work on Apple M1.

**Comment by joreilly** [comment]
looks like `watchosX64` support is gradually making it in to different libraries e.g. https://github.com/cashapp/sqldelight/commit/4e2d41087d38e367ad41fb659b991b3844ff9174 and @RdeWilde added to Koin (https://github.com/InsertKoinIO/koin/commit/4d90432aa792ed55d6b113c82c9a677fb68205e5) but guess not in released version yet?

**Comment by RdeWilde** [comment]
Correct @joreilly . I opened issues on all the dependencies that were not yet supporting it. Some have implemented it, but it is still only in the master branch, not into a release. I guess it'll will need some time or you have to rely on unstable 'releases'. which is not really a good option I think.

**Comment by joreilly** [comment]
https://github.com/cashapp/sqldelight/releases/tag/1.5.1

"[Native Driver] Add watchosX64 support"

**Comment by joreilly** [comment]
Ktor 1.6.1 was supposed to have watchosX64 support but I see in https://youtrack.jetbrains.com/issue/KTOR-2678

"Because of the bug in our release automation, watchosX64 was not published. It will be fixed in the next release. For now, you can use EAP, which should have this target published."

**Comment by RdeWilde** [comment]
https://github.com/ktorio/ktor/pull/2511#issuecomment-884832608 ETA end of July

**Comment by RdeWilde** [comment]
It seems to include all the commits upto now, so also this one:
https://github.com/ktorio/ktor/pull/2511/files

I'll give it a try.

**Comment by RdeWilde** [comment]
@joreilly Seems to be succesfull! I tested with Gradle 7.0.0 and Ktor 1.6.2 and SQLdelight 1.5.1  ✅

**Comment by joreilly** [comment]
Hi @RdeWilde , are you still hoping to create MR with your changes?

**Comment by joreilly** [comment]
Hi @RdeWilde, sorry to ping you again about this but be great if possible to get this resolved.....do you think you'll get chance to create PR.....no worries if not, I can probably try and see what's needed.
